docs/core/porting/cpp-cli-alts.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Migrating C++/CLI projects to ISO C++ and/or .NET
+description: Learn about alternatives to C++/CLI
+author: AaronRobinsonMSFT
+ms.date: 01/30/2023
+---
+
+# Migration from C++/CLI to ISO C++ and/or C\#
+
+The C++/CLI language has been around since 2005 and has enabled users to easily integrate C++ and a .NET language into a single binary. The utility of C++/CLI remains as an innovative way to integrate C++ with the .NET runtime on the Windows platform. There are however limitations to using C++/CLI and as ISO C++ and .NET platforms evolve these limitations are expected to grow.
+
+**Current Limitations**
+
+- Only operates on the Windows platform.
+
+- Limited functional support for the latest ISO C++.
+
+- Limited support for newer .NET types and patterns (for example, [`Span<T>`](/dotnet/api/system.span-1) and [default interface methods](/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods)).
+
+If the above limitations are a concern then this document is intended to provide options on how to transtition from C++/CLI to ISO C++ and/or .NET. The approaches described represent a preference for cross-platform and access to the respective language's entire feature set.
+
+## Expose C++ as C exports
+
+.NET has supported directly calling C code since .NET Framework 1.0 through the [Platform Invoke](/dotnet/standard/native-interop/pinvoke) (P/Invokes) mechanism. All C++ APIs that are expected to be called from .NET could be wrapped in a C-style function. This approach has always worked with .NET but prior to .NET 7, the recommendation was to use [`DllImportAttribute`](/dotnet/api/system.runtime.interopservices.dllimportattribute). Since .NET 7, the [`LibraryImportAttribute`](/dotnet/standard/native-interop/pinvoke-source-generation?source=recommendations) should be prefered in all cases. As it relates to C++/CLI, the `LibraryImportAttribute` mechanism provides a higher performance solution to calling C exports than previously available. Furthermore, the extensibility provided by `LibraryImportAttribute` enables developers to create customized marshalling schemes where they are in full control of how and when data is marshalled across the interop boundary. This is in stark contrast to using `DllImportAttribute` that imposes significant performance penalties if custom marshalling is needed.
```

Comment:
> supported since .NET Framework 1.0

I get it, but it is unnecessary to provide this historical context.

```suggestion
.NET supports directly calling C code through the [Platform Invoke](/dotnet/standard/native-interop/pinvoke) (P/Invokes) mechanism. All C++ APIs that are expected to be called from .NET could be wrapped in a C-style function. This approach has always worked with .NET but prior to .NET 7, the recommendation was to use [`DllImportAttribute`](/dotnet/api/system.runtime.interopservices.dllimportattribute). Since .NET 7, the [`LibraryImportAttribute`](/dotnet/standard/native-interop/pinvoke-source-generation?source=recommendations) should be prefered in all cases. As it relates to C++/CLI, the `LibraryImportAttribute` mechanism provides a higher performance solution to calling C exports than previously available. Furthermore, the extensibility provided by `LibraryImportAttribute` enables developers to create customized marshalling schemes where they are in full control of how and when data is marshalled across the interop boundary. This is in stark contrast to using `DllImportAttribute` that imposes significant performance penalties if custom marshalling is needed.
```